src/app/services/courses-store.service.ts

```diff
@@ -1,44 +1,162 @@
+import { Author } from '@app/shared/types/author.model';
 
 @Injectable({
-  providedIn: "root",
+  providedIn: 'root',
 })
 export class CoursesStoreService {
+  private courses$$ = new BehaviorSubject<Course[]>([]);
+  public courses$ = this.courses$$.asObservable();
+  private authors$$ = new BehaviorSubject<Author[]>([]);
+  public authors$ = this.authors$$.asObservable();
+  private isLoading$$ = new BehaviorSubject<boolean>(false);
+  public isLoading$ = this.isLoading$$.asObservable();
+
+  constructor(private courseService: CoursesService) {}
+
   getAll() {
-    // Add your code here
+    this.isLoading$$.next(true);
+    this.courseService.getAll().subscribe({
+      next: (response: AllCourseResponse) => {
+        this.courses$$.next(response.result);
+        this.isLoading$$.next(false);
+      },
+      error: err => {
+        console.error('Failed to get courses', err);
+        this.isLoading$$.next(false);
+      },
+    });
   }
 
-  createCourse(course: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  createCourse(course: Omit<Course, 'id'>) {
+    this.isLoading$$.next(true);
+    this.courseService.createCourse(course).subscribe({
+      next: (response: CreateCourseResponse) => {
+        this.courses$$.next([...this.courses$$.value, response.result]);
+        this.isLoading$$.next(false);
+      },
+      error: err => {
+        this.isLoading$$.next(false);
+        console.error('Failed to add course', err);
+      },
+    });
   }
 
-  getCourse(id: string) {
-    // Add your code here
+  getCourse(id: string): Observable<CourseResponse> {
+    this.isLoading$$.next(true);
+    return this.courseService.getCourse(id).pipe(
+      tap({
+        next: () => {
+          this.isLoading$$.next(false);
+        },
+        error: err => {
+          console.error('Failed to get course', err);
+          this.isLoading$$.next(false);
+        },
+      })
+    );
   }
 
-  editCourse(id: string, course: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  editCourse(id: string, course: Omit<Course, 'id' | 'creationDate'>) {
+    this.isLoading$$.next(true);
+    this.courseService.editCourse(id, course).subscribe({
+      next: () => {
+        this.courses$$.next(
+          this.courses$$.value.map(courseElement => {
+            if (courseElement.id === id) return { ...course, id: id };
+            else return courseElement;
+          })
+        );
+        this.isLoading$$.next(false);
+      },
+      error: err => {
+        this.isLoading$$.next(false);
+        console.error('Failed to edit course', err);
+      },
+    });
   }
 
   deleteCourse(id: string) {
-    // Add your code here
+    this.isLoading$$.next(true);
+    this.courseService.deleteCourse(id).subscribe({
+      next: (response: DeleteCourseResponse) => {
+        if (response.successful) this.courses$$.next(this.courses$$.value.filter(course => course.id !== id));
```

Comment:
Like above, get the fresh list from API

Reply:
after delete we get the courses and authors too.
Done

Reply:
Ok, so we have a successful response. I think it is no longer necessary to manually assign new value to the `courses$$`, since you are triggering a new call on the next line. Please check this here and all other occurrences with course, author methods.
![image](https://github.com/user-attachments/assets/2433f783-2d4e-4036-8e16-0053f4581fdd)

Reply:
Totally right! Good point